src/ast.c

```diff
@@ -731,6 +731,7 @@ jl_value_t *jl_parse_eval_all(const char *fname, size_t len,
         fl_free_gc_handles(fl_ctx, 1);
     }
     else {
+        assert(memchr(fname, 0, len) == NULL); // was checked already in jl_load
```

Comment:
On release-0.4, `jl-parse-file` is used inside `jl_start_parsing_file` which does not take the `len` as input. Could you propose a backport, preferably w.r.t. tk/backports-0.4.6 if you want to get this in?